### PR TITLE
added solution 2.2

### PR DESCRIPTION
**Pull request by TimoxaV**

### CODE REVIEW

File: src/main/java/core/basesyntax/Machine.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public abstract class Machine {
+
+    String machineTitle;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Machine.java:5]: let's add access modifier

File: src/main/java/core/basesyntax/MainApp.java [hunk 2]
```diff
@@ -12,5 +12,17 @@
  * - Отправить все машини на работу и остановить работу всех машин.
  */
 public class MainApp {
+    public static void main(String[] args) {
+        Track track = new Track("Track001");
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MainApp.java:16]: be attentive with [checklist](https://mate-academy.github.io/jv-program-fulltime/01_core/checklist/hw_02.html)
- Use abstract class to create Track, Bulldozer and Excavator instances.

Example:
Animal cat = new Cat();

Comment (TimoxaV) [reply src/main/java/core/basesyntax/MainApp.java]: done

File: src/main/java/core/basesyntax/Machine.java (hunk 1 above)
Comment (TimoxaV) [reply src/main/java/core/basesyntax/Machine.java]: done

File: src/main/java/core/basesyntax/Machine.java [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 public abstract class Machine {
 
-    String machineTitle;
+    protected String machineTitle;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Machine.java:5]:
```suggestion
    private String machineTitle;
```
it is one of the OOP principal: encapsulation

Comment (TimoxaV) [reply src/main/java/core/basesyntax/Machine.java]: Done. And added getter for machineTitle.

File: src/main/java/core/basesyntax/Bulldozer.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+public class Bulldozer extends Machine {
+    public Bulldozer(String machineTitle) {
+        super(machineTitle);
+    }
+
+    @Override
+    public void doWork() {
+        System.out.println(this.getMachineTitle() + " is working!");
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Bulldozer.java:10]:
```suggestion
        System.out.println(getMachineTitle() + " is working!");
```

File: src/main/java/core/basesyntax/Bulldozer.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    }
+
+    @Override
+    public void stopWork() {
+        System.out.println(this.getMachineTitle() + " is doing nothing!");
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Bulldozer.java:15]:
```suggestion
        System.out.println(getMachineTitle() + " is doing nothing!");
```

File: src/main/java/core/basesyntax/Bulldozer.java (hunk 4 above)
Comment (TimoxaV) [reply src/main/java/core/basesyntax/Bulldozer.java]: done

File: src/main/java/core/basesyntax/Bulldozer.java (hunk 5 above)
Comment (TimoxaV) [reply src/main/java/core/basesyntax/Bulldozer.java]: done

**Pull request closed without merging**